api/src/utils/apply-query.ts

```diff
@@ -260,11 +260,13 @@ export function applySort(
 	knex: Knex,
 	schema: SchemaOverview,
 	rootQuery: Knex.QueryBuilder,
-	rootSort: string[],
+	query: Query,
 	collection: string,
 	aliasMap: AliasMap,
 	returnRecords = false
 ) {
+	const rootSort = query.sort!;
```

Comment:
Do we need the non-null assertion here? or could we use an early return instead 🤔

Reply:
While usually the answer is _absolutely yes_, in this case the one place where this is called already explicitly has that check, so I think the not-null assertion here is acceptable:

https://github.com/directus/directus/blame/c0218b471737ab9a39961e147bca587d1fca285d/api/src/utils/apply-query.ts#L53

(also with the knowledge that this whole fire is being replaced with a way more strict implementation as we speak 😁 )